test/test_autograd.py

```diff
@@ -5336,6 +5336,42 @@ def assert_only_first_requires_grad(res):
                                          return_counts=return_counts)
                     assert_only_first_requires_grad(res)
 
+    def test_custom_function_cycle(self):
+        class MyFn(Function):
+            @staticmethod
+            def forward(ctx, x):
+                x = x.clone()
+                # BAD!!! Don't do this ever!!
```

Comment:
You should explain why it's OK to do this in the test, and under what circumstances the test should be "fixed" if this behavior breaks in the future

Reply:
It would be nicer if we had a real world example of a loop that was resolved this way.